src/wakefs/fuse.cpp

```diff
@@ -217,9 +227,35 @@ bool run_in_fuse(fuse_args &args, int &status, std::string &result_json) {
   (void)close(STDOUT_FILENO);
   (void)close(STDERR_FILENO);
 
-  do waitpid(pid, &status, 0);
-  while (WIFSTOPPED(status));
+  int pidfd = syscall(SYS_pidfd_open, pid, 0);
```

Comment:
`SYS_pidfd_open` was added in linux kernel 5.3 which is a bit more recent than some of the systems we try to support. I suspect that impacted trying to find a libc wrapper too

Reply:
I was poking around and this seemed like the only way to do it with epoll. @JakeSiFive any alternative ideas? If not we might have to just use sleep

Reply:
What does /usr/bin/timeout do? We could even just add it to the `execve_wrapper` to prepend it?

Reply:
I think it probably uses something like timer create but I'll go read the source code tomorrow. https://man7.org/linux/man-pages/man2/timer_create.2.html

Reply:
I guess you could do this:

1) Create your payload process
2) Create a timer process that sleeps and then exits
3) wait on both of them, if the timer process exits first then you kill the payload process